hd-docs/system-overview/feature-map.md

```
@@ -1,142 +1,420 @@
+**Category:** LLM Providers  
+**Files:** `/server/utils/AiProviders/openAi/`, `/server/utils/AiProviders/anthropic/`, `/server/utils/AiProviders/azureOpenAi/`, etc.  
+**Technologies/Libraries:** OpenAI SDK, Anthropic SDK, Azure OpenAI, AWS Bedrock, Google Gemini  
+**How it works:** Extensive support for commercial LLM providers including OpenAI (GPT models), Anthropic (Claude), Azure OpenAI, AWS Bedrock, Google Gemini, Cohere, Groq, Perplexity, OpenRouter, DeepSeek, Mistral, xAI, and many others. Each provider has dedicated integration with API key management and model selection.
+
+### **Local LLM Support**
+**Category:** LLM Providers  
+**Files:** `/server/utils/AiProviders/ollama/`, `/server/utils/AiProviders/lmStudio/`, `/server/utils/AiProviders/localAi/`, `/server/utils/AiProviders/textGenWebUI/`  
+**Technologies/Libraries:** Ollama, LM Studio, LocalAI, Text Generation WebUI, KoboldCPP  
+**How it works:** Comprehensive local LLM support through popular local inference servers. Includes automatic model discovery, endpoint configuration, and seamless integration with both local and cloud models.
+
+### **Model Context Protocol (MCP) Support**
```

markdownlint-cli2 (0.17.2)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)